presto-main/src/main/java/com/facebook/presto/operator/BigintGroupByHash.java

```diff
@@ -327,4 +320,109 @@ private static int calculateMaxFill(int hashSize)
         checkArgument(hashSize > maxFill, "hashSize must be larger than maxFill");
         return maxFill;
     }
+
+    private class AddPageWork
+            implements Work<Void>
+    {
+        private final Block block;
+
+        private boolean finished;
+        private int lastPosition;
+
+        public AddPageWork(Block block)
+        {
+            this.block = requireNonNull(block, "block is null");
+        }
+
+        @Override
+        public boolean process()
+        {
+            if (finished) {
+                return true;
+            }
+
+            if (needRehash() && !tryRehash()) {
+                return false;
+            }
+
+            int positionCount = block.getPositionCount();
+            verify(lastPosition < positionCount, "position count out of bound");
+            while (lastPosition < positionCount) {
+                // get the group for the current row
+                putIfAbsent(lastPosition, block);
+                lastPosition++;
+
+                if (needRehash()) {
+                    // no need to try rehashing; putIfAbsent should have done that
+                    break;
+                }
+            }
+            finished = lastPosition == positionCount;
+            return finished;
+        }
+
+        @Override
+        public Void getResult()
+        {
+            throw new UnsupportedOperationException(getClass().getName());
```

Comment:
This is effectively final class, so just `throw new UnsupportedOperationException()`